components/post_view/post_view.jsx

```diff
@@ -0,0 +1,393 @@
+import {getNewestPostIdFromPosts, getOldestPostIdFromPosts} from 'mattermost-redux/utils/post_utils';
+
+import LoadingScreen from 'components/loading_screen.jsx';
+import CreateChannelIntroMessage from 'components/post_view/channel_intro_message';
+import Constants, {PostRequestTypes} from 'utils/constants.jsx';
+
+import PostList from './post_list.jsx';
+
+const POSTS_PER_PAGE = Constants.POST_CHUNK_SIZE / 2;
+
+export default class PostView extends React.PureComponent {
+    static propTypes = {
+
+        /**
+         * Array of posts in the channel, ordered from newest to oldest
+         */
+        posts: PropTypes.array,
+
+        /**
+         * The channel the posts are in
+         */
+        channelId: PropTypes.string,
+
+        /**
+         * Used for disabling loading more messages
+         */
+        postVisibility: PropTypes.number,
+
+        /**
+         * Used for showing more message indicators and channel intro message
+         */
+        channelPostsStatus: PropTypes.object,
+
+        /**
+         * Used for unread count
+         */
+        currentUserId: PropTypes.string,
+
+        /**
+         * Used for unread count
+         */
+        lastViewedAt: PropTypes.number,
+
+        /**
+         * Flag used to determine if channel has to call sync actions.
+         */
+        channelSyncStatus: PropTypes.bool,
+
+        /**
+         * Used for websocket connect change to trigger sync.
+         */
+        socketStatus: PropTypes.object,
+
+        /**
+         * Used for determining if we reached to the top of channel.
+         * Used for taking a backup before clearing from redux-store.
+         */
+        postIdsInCurrentChannel: PropTypes.array,
+
+        focusedPostId: PropTypes.string,
+
+        actions: PropTypes.shape({
+
+            /*
+             * Get post for permaview
+             */
+            getPostThread: PropTypes.func.isRequired,
+
+            /*
+             * Get unreads posts onload
+             */
+            loadUnreads: PropTypes.func.isRequired,
+
+            /*
+             * Get posts using BEFORE_ID and AFTER_ID
+             */
+            loadPosts: PropTypes.func.isRequired,
+
+            /*
+             * Used for changing posts status of channel
+             * For notifying redux store when we hit bottom or top of posts
+             */
+            changeChannelPostsStatus: PropTypes.func.isRequired,
+
+            /*
+             * Used for notifying redux store on channel sync complete
+             */
+            channelSyncCompleted: PropTypes.func.isRequired,
+
+            /*
+             * Used to sync when channel is out of sync because of socket disconnect
+             */
+            syncChannelPosts: PropTypes.func.isRequired,
+
+        }).isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            isDoingInitialLoad: this.shouldLoadPosts(props),
```

Comment:
To continue [this conversation](https://github.com/mattermost/mattermost-webapp/pull/1464#discussion_r206205929), why can't we just check `this.shouldLoadProps` in the render method? We have
```javascript
constructor(props) {
    this.state = {
        isDoingInitialLoad: this.shouldLoadPosts(props),
    };
}

render() {
    if (posts.length === 0 && this.state.isDoingInitialLoad) {
         return <LoadingScreen/>;
    }

    return ...;
}
```
but we could get rid of some of the state and just do
```javascript
render() {
    if (posts.length === 0 && this.shouldLoadPosts(this.props)) {
         return <LoadingScreen/>;
    }

    return ...;
}
```

The same thing applies to some of the other values that we're storing in this.state.

Reply:
Removed isDoingInitialLoad and lastViewedAt from state.
But other state variables cannot be moved to render as they depend on api call response or initiation.

```
newerPosts : {
   loading: this.value is derived from api call trigger(So, i dont see how we can check this on render).
   allLoaded: this is based on the response of api call. 
}
```


Similar case with `olderPosts` object in state.